activity_browser/app/ui/graphics.py

```diff
@@ -56,104 +50,123 @@ def __init__(self, parent, data, labels, dpi=100):
         vmax = np.abs(corr.values[~mask]).max()
         # vmax = np.abs(corr).max()
         sns.heatmap(corr, mask=mask, cmap=plt.cm.PuOr, vmin=-vmax, vmax=vmax,
-                    square=True, linecolor="lightgray", linewidths=1, ax=axes)
+                    square=True, linecolor="lightgray", linewidths=1, ax=self.ax)
         for i in range(len(corr)):
-            axes.text(i + 0.5, i + 0.5, corr.columns[i],
+            self.ax.text(i + 0.5, i + 0.5, corr.columns[i],
                       ha="center", va="center",
                       rotation=0 if len(labels) <= 8 else 45,
                       size=11 if len(labels) <= 8 else 9)
             for j in range(i + 1, len(corr)):
                 s = "{:.3f}".format(corr.values[i, j])
-                axes.text(j + 0.5, i + 0.5, s,
+                self.ax.text(j + 0.5, i + 0.5, s,
                           ha="center", va="center",
                           rotation=0 if len(labels) <= 8 else 45,
                           size=11 if len(labels) <= 8 else 9)
-        axes.axis("off")
-        # If uncommented, fills widget
-        self.setSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
-        self.updateGeometry()
-        self.setMinimumSize(self.size())
+        self.ax.axis("off")
 
+        # refresh canvas
+        self.canvas.draw()
+        size = self.figure.get_size_inches() * self.figure.dpi
+        self.setMinimumSize(size[0], size[1])
+
+
+class LCAResultsPlot(Plot):
+    def __init__(self, parent=None, *args):
+        super(LCAResultsPlot, self).__init__(parent, *args)
+
+    def plot(self, mlca):
+        # need to clear the figure and add axis again
+        # because of the colorbar which does not get removed by the ax.clear()
+        self.figure.clf()
+        self.ax = self.figure.add_subplot(111)
 
-class LCAResultsPlot(FigureCanvasQTAgg):
-    def __init__(self, parent, mlca, width=6, height=6, dpi=100):
         activity_names = [
             format_activity_label(next(iter(f.keys())), style='pnl') for f in mlca.func_units
         ]
-        figure = Figure(figsize=(2+len(mlca.methods)*0.5, 4+len(activity_names)*0.55),
-                        dpi=dpi, tight_layout=True)
-        axes = figure.add_subplot(111)
+        size = (2 + len(mlca.methods) * 0.5, 4 + len(activity_names) * 0.55)
+        self.figure.set_size_inches(size[0], size[1])
```

Comment:
General question: Where are the figures scaled to fit into the right panel, if we give a fixed size in inches here? I tested it with lower display resolution and it still looks fine. Will try with a 4k screen when I find the time.

Reply:
I think the tight_layout=True does the trick. This requires the latest version of matplotlib (2.2.2).